bitnami/spark/templates/deployment-master.yaml

```diff
@@ -0,0 +1,183 @@
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - name: http
+              containerPort: {{ .Values.master.webPort }}
+              protocol: TCP
+            - name: cluster
+              containerPort: {{ .Values.master.clusterPort }}
+          volumeMounts:
+          {{- if .Values.master.configurationConfigMap }}
+            - name: config
+              mountPath: '/opt/bitnami/spark/conf/'
+          {{- end }}
+          {{- if .Values.security.certificatesSecretName }}
+            - name: certs
+              mountPath: '/opt/bitnami/spark/conf/certs'
+              readOnly: true
+          {{- end }}
+          env:
+          - name: SPARK_MODE
+            value: "master"
+          {{- if .Values.master.debug }}
+          - name: BASH_DEBUG
+            value: "1"
+          - name: NAMI_DEBUG
+            value: "1"
+          - name: NAMI_LOG_LEVEL
+            value: "trace8"
+          {{- end }}
+          - name: SPARK_DAEMON_MEMORY
+            value: {{ .Values.master.daemonMemoryLimit | quote }}
+          {{- if .Values.master.clusterPort }}
+          - name: SPARK_MASTER_PORT
+            value: {{ .Values.master.clusterPort | quote }}
+          {{- end }}
+          {{- if .Values.master.webPort }}
+          - name: SPARK_MASTER_WEBUI_PORT
+            value: {{ .Values.master.webPort | quote }}
+          {{- end }}
+          - name: SPARK_MASTER_OPTS
+            value: {{ .Values.master.configOptions | quote }}
+          {{- if eq ( .Values.security.rpc.authenticationEnabled | quote ) "yes" }}
+          - name: SPARK_RPC_AUTHENTICATION_ENABLED
+            value: {{ .Values.security.rpc.authenticationEnabled | quote }}
+          - name: SPARK_RPC_AUTHENTICATION_SECRET
+            valueFrom:
+              secretKeyRef:
+              {{- if .Values.security.passwordsSecretName }}
+                name: {{ .Values.security.passwordsSecretName }}
+                key: rpc-authentication-secret
+              {{- else }}
+                name: {{ include "spark.fullname" . }}-secret
+                key: rpc-authentication-secret
+              {{- end }}
+          {{- end }}
+          {{- if eq ( .Values.security.encryptionEnabled | quote ) "yes" }}
+          - name: SPARK_RPC_ENCRYPTION_ENABLED
+            value: {{ .Values.security.rpc.encryptionEnabled | quote }}
+          {{- end }}
+          {{- if eq ( .Values.security.storageEncriptionEnabled | quote ) "yes" }}
+          - name: SPARK_LOCAL_STORAGE_ENCRYPTION_ENABLED
+            value: {{ .Values.security.storageEncryptionEnabled | quote }}
+          {{- end }}
+          {{- if eq ( .Values.security.ssl.enabled | quote ) "yes" }}
+          - name: SPARK_SSL_ENABLED
+            value: {{ .Values.security.ssl.enabled | quote }}
+          - name: SPARK_SSL_KEY_PASSWORD
+            valueFrom:
+              secretKeyRef:
+              {{- if .Values.security.passwordsSecretName }}
```

Comment:
Could you move this section inside a template? This is useful for situations where an umbrella chart wants to add secrets

Example: https://github.com/helm/charts/blob/master/stable/postgresql/templates/_helpers.tpl#L240

Reply:
Yes, it is a better approach to not replicate the code. Thanks for the advice. Done.